stock_barcodes/wizard/stock_barcodes_read_picking.py

```diff
@@ -0,0 +1,343 @@
+        if len(self.candidate_picking_ids) == 1:
+            return self.candidate_picking_ids.picking_id
+        else:
+            return self.env['stock.picking'].browse()
+
+    def check_done_conditions(self):
+        res = super().check_done_conditions()
+        if self.product_id.tracking != 'none' and not self.lot_id:
+            self._set_messagge_info('info', _('Waiting for input lot'))
+            return False
+        if not self.picking_id:
+            if not self._search_candidate_pickings():
+                self._set_messagge_info(
+                    'info', _('Click on picking pushpin to lock it'))
+                return False
+        if self.picking_id != self._candidate_picking_selected():
+            self._set_messagge_info(
+                'info', _('Click on picking pushpin to lock it'))
+            return False
+        return res
+
+    def _prepare_scan_log_values(self, log_detail=False):
+        # Store in read log line each line added with the quantities assigned
+        vals = super()._prepare_scan_log_values(log_detail=log_detail)
+        vals['picking_id'] = self.picking_id.id
+        if log_detail:
+            vals['log_line_ids'] = [(0, 0, {
+                'move_line_id': x[0],
+                'product_qty': x[1],
+            }) for x in log_detail.items()]
+        return vals
+
+    def remove_scanning_log(self, scanning_log):
+        for log in scanning_log:
+            for log_scan_line in log.log_line_ids:
+                if log_scan_line.move_line_id.state not in ['assigned',
+                                                            'confirmed']:
+                    raise ValidationError(_(
+                        'You can not remove an entry linked to a stock move '
+                        'line in state assigned or confirmed')
+                    )
+                qty = (log_scan_line.move_line_id.qty_done -
+                       log_scan_line.product_qty)
+                log_scan_line.move_line_id.qty_done = max(qty, 0.0)
+            self.picking_product_qty = sum(log.log_line_ids.mapped(
+                'move_line_id.move_id.quantity_done'))
+            log.unlink()
+
+    def action_undo_last_scan(self):
+        res = super().action_undo_last_scan()
+        log_scan = first(self.scan_log_ids.filtered(
+            lambda x: x.create_uid == self.env.user))
+        self.remove_scanning_log(log_scan)
+        return res
+
+
+class WizCandidatePicking(models.TransientModel):
+    """
+    TODO: explain
+    """
+    _name = 'wiz.candidate.picking'
+    _description = 'Candidate pickings for barcode interface'
+    # To prevent remove the record wizard until 2 days old
+    _transient_max_hours = 48
+
+    wiz_barcode_id = fields.Many2one(
+        comodel_name='wiz.stock.barcodes.read.picking',
+        readonly=True,
+    )
+    picking_id = fields.Many2one(
+        comodel_name='stock.picking',
+        string='Picking',
```

Comment:
Odoo will call it from the name of the field. You do not need to define the string name.